templates/aws-vpc.template.yaml

```diff
@@ -340,6 +395,29 @@ Conditions:
   PrivateSubnets&4AZCondition: !And
     - !Condition 'PrivateSubnetsCondition'
     - !Condition '4AZCondition'
+  PrivateSubnets&PublicSubnets&NatGatewaysCondition: !And
```

Comment:
I think this condition is redundant. You really don't need this because you are already checking that if nat gateway condition is true than private and public subnet condition must be true as well, in the assert rule above. So, If you just use natgatewaycondition where u r using this condition, it will have the same effect.